### ISSUE TITLE
Supporting Gulp?

### ISSUE THREAD

**Issue by mtraynham** [allegro/grunt-maven-plugin#79]
Gulp has quite a few benefits over Grunt, specifically that it's built on a streaming concept.  Any chance that this may get added?

**Comment by adamdubiel** [comment]
Core concepts would be the same, so simply adding it should be quite simple. I don't know about some extended support. Never used Gulp (yet), so i'm not so familiar with this tool. Let me think about it, i need to assess how much time would it take to make it right.

**Comment by mtraynham** [comment]
Sure thing.  I suspect that the grunt-maven npm package would not be reusable, but the `grunt-maven.json` file would be.

**Comment by adamdubiel** [comment]
Exactly. I'm not convinced i want to go though writing custom gulp plugin (or maybe it's easier than in grunt?). One step at a time.

**Comment by mtraynham** [comment]
I imagine just triggering Gulp through Maven instead of Grunt is probably enough.    In terms of the `grunt-maven` npm module, it has the tasks:
- mavenPrepare - copies resources from `src` to `target-grunt`
- mavenDist - copies resources from `target-grunt` -> `target-grunt/dist` -> `target`

`mavenPrepare` is very nice when using a watch task, but I'm under the assumption that the `mavenDist` task is not all particularly useful.  I've superseded its use with the maven assembly plugin instead, as Grunt already builds the *deliverables* and I just tar/collect only the files I need.

So question, based on my usages of the npm package (which seems somewhat optional), could they not be replaced by something like [grunt-sync](https://github.com/tomusdrw/grunt-sync) or with gulp, [it supports it out of the box](http://blog.andrewray.me/how-to-copy-only-changed-files-with-gulp/).

Not trying to deter you from dropping support for the tasks, I'm just wondering if they are necessary to the end-to-end workflow.

**Comment by adamdubiel** [comment]
I was just thinking of starting small - allowing to run Gulp. The rest will come if needed :)

I started out as grunt-only plugin, but now i see that pivoting to more agnostic plugin might be nice.

**Comment by mtraynham** [comment]
Awesome.  Alternatively, I was able to get this running gulp, by setting:
```
<!-- Using gulp instead -->
<gruntExecutable>gulp</gruntExecutable>
```

Although, seems a bit like a hack.
